[
  {
    "number": 159,
    "title": "libprotobuf ERROR while running sentiment demo",
    "created_at": "2016-10-04T23:22:09Z",
    "closed_at": "2016-10-06T04:29:57Z",
    "labels": [
      "question"
    ],
    "url": "https://github.com/PaddlePaddle/Paddle/issues/159",
    "body": "1. paddle version\n   git version: commit 29c16e22420b8b69d17572accd2f6fc0239c8c7e\n   paddle compiled from source, with the options: cmake .. -DWITH_GPU=ON -DWITH_DOC=ON -DWITH_SWIG_PY=ON -DCMAKE_INSTALL_PREFIX=/data/user/program/pddl_build/\n   make and make install success.\n2. protoc --version\n   libprotoc 2.5.0\n3. run the sentiment demo:\n   cd /demo/sentiment/data/\n   ./get_imdb.sh\n   cd ..\n   ./preprocess.sh\n   the above steps are good.\n   However, run ./train.sh, I get the following error msg:\n# \n\nI1004 16:22:13.414708 12100 Util.cpp:144] commandline: /usr/local/bin/../opt/paddle/bin/paddle_trainer --config=trainer_config.py --save_dir=./model_output --job=train --use_gpu=false --trainer_count=4 --num_passes=10 --log_period=10 --dot_period=20 --show_parameter_stats_period=100 --test_all_data_in_one_period=1 \nI1004 16:22:13.414949 12100 Util.cpp:113] Calling runInitFunctions\nI1004 16:22:13.415261 12100 Util.cpp:126] Call runInitFunctions done.\n[INFO 2016-10-04 16:22:13,802 networks.py:1125] The input order is [word, label]\n[INFO 2016-10-04 16:22:13,802 networks.py:1132] The output order is [**cost_0**]\n[libprotobuf ERROR google/protobuf/message_lite.cc:123] Can't parse message of type \"paddle.TrainerConfig\" because it is missing required fields: model_config.parameters[0].learning_rate, model_config.parameters[0].momentum, model_config.parameters[1].learning_rate, model_config.parameters[1].momentum, model_config.parameters[2].learning_rate, model_config.parameters[2].momentum, model_config.parameters[3].learning_rate, model_config.parameters[3].momentum, model_config.parameters[4].learning_rate, model_config.parameters[4].momentum, model_config.parameters[5].momentum, model_config.parameters[6].momentum, model_config.parameters[7].learning_rate, model_config.parameters[7].momentum, model_config.parameters[8].learning_rate, model_config.parameters[8].momentum, model_config.parameters[9].learning_rate, model_config.parameters[9].momentum, model_config.parameters[10].momentum, model_config.parameters[11].momentum, model_config.parameters[12].learning_rate, model_config.parameters[12].momentum, model_config.parameters[13].learning_rate, model_config.parameters[13].momentum, model_config.parameters[14].learning_rate, model_config.parameters[14].momentum, model_config.parameters[15].momentum, model_config.parameters[16].momentum, model_config.parameters[17].learning_rate, model_config.parameters[17].momentum\nF1004 16:22:13.825393 12100 TrainerConfigHelper.cpp:59] Check failed: m->conf.ParseFromString(configProtoStr) \n**\\* Check failure stack trace: ***\n    @     0x7fc3d0879daa  (unknown)\n    @     0x7fc3d0879ce4  (unknown)\n    @     0x7fc3d08796e6  (unknown)\n    @     0x7fc3d087c687  (unknown)\n    @           0x6aa559  paddle::TrainerConfigHelper::TrainerConfigHelper()\n    @           0x6aa924  paddle::TrainerConfigHelper::createFromFlags()\n    @           0x53af53  main\n    @     0x7fc3cfa85ec5  (unknown)\n    @           0x546695  (unknown)\n    @              (nil)  (unknown)\n# /usr/local/bin/paddle: line 46: 12100 Aborted                 (core dumped) ${DEBUGGER} $MYDIR/../opt/paddle/bin/paddle_trainer ${@:2}\n\nCould you please help to check what is this about?\n",
    "comments_url": "https://api.github.com/repos/PaddlePaddle/Paddle/issues/159/comments",
    "author": "cactiball",
    "comments": [
      {
        "user": "emailweixu",
        "created_at": "2016-10-05T03:49:54Z",
        "body": "It looks like your conflict version of paddle c++ binary and python interface. Your install prefix is different from the path shown in the above log\n"
      },
      {
        "user": "cactiball",
        "created_at": "2016-10-05T06:42:51Z",
        "body": "Thank you very much! Append the install prefix to PATH in ~/.bashrc solved this issue.\n"
      }
    ],
    "satisfaction_conditions": [
      "Resolve path/environment configuration mismatch between compiled binaries and execution environment",
      "Ensure version compatibility between Paddle's C++ components and Python interface"
    ],
    "_classification": {
      "category": "Can be dockerized without any issue",
      "timestamp": "2025-04-05 00:37:10"
    }
  }
]